app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -359,12 +451,36 @@ class PipelineViz extends React.Component {
   }
 
   handleNodeBlur = () => {
-    this.graphs.forEach(graph => {
-      const allEdges = this.getEdgesBetweenNodes(graph, null, null, "colored");
-      graph.updateEdges(allEdges, { hidden: true });
+
+    const { nodeColor } = this.props;
+    const { graphIndex, nodeId, alteredGraphs } = this.lastMouseMoveInfo;
+
+    if (nodeId == null) {
+      return;
+    }
+
+    const defaultNodeOptions = {
+      borderWidth: 1,
+      color: {
+        background: nodeColor,
+        border: nodeColor,
+      },
+    };
+    this.graphs[graphIndex].updateNodes([nodeId], defaultNodeOptions);
+
+    alteredGraphs.forEach(i => {
+      const graph = this.graphs[i];
+      const allColoredEdgeIds = this.getEdgesBetweenNodes(
```

Comment:
nit: since you're often filing the params to getEdgesBetweenNodes with 1-2 nulls, you could consider making the params an object and checking for fromId, toId